src/index.spec.ts

```diff
@@ -871,6 +932,7 @@ describe('initialization options', () => {
         assignmentLogger: mockLogger,
         updateOnFetch,
         useExpiredCache: true,
+        forceReinitialize: true,
```

Comment:
some of the test cases here could be incorrect. this one:

```
// Init again with an expired cache so a fetch will kicked off too
// We allow the expired cache, so it will succeed and fetch will be delayed
```

might be contrary to the singleton approach - or perhaps we need to check more conditions than just `initialized`

Reply:
I think we were using multiple initializations as a hack to check populating and then using the cache. If we want, we could rework tests to check cache contents and set up an already-populated cache another way but I think this is probably fine for now.

Reply:
ah nvm it looks like we use pino exactly for that